### PR TITLE
Fix bug:getTopology will be failed on Arm platform

### PR DESCRIPTION
**Pull request by lubinszARM**
On Arm platform, no 'core id' and 'physical id' in '/proc/cpuinfo'.
So we should search sysfs cpu path directly to get the data of
'thread_id' &'core_id' & 'node_id'.
This method can also be used on other platforms, such as x86, ppc64le...
/sys/bus/cpu/devices/cpu%d contains the information of 'core_id' & 'node_id'.
Such as:
cat /sys/bus/cpu/devices/cpu0/topology/core_id
ls /sys/bus/cpu/devices/cpu0/node0

Signed-off-by: Bin Lu <bin.lu@arm.com>

### CODE REVIEW

**Comment by lubinszARM** [comment]
@dashpole 
Hi, I have fixed the CLA issue.
Any comments on this?

File: machine/machine.go [hunk 1]
```diff
@@ -127,6 +130,60 @@ func parseCapacity(b []byte, r *regexp.Regexp) (uint64, error) {
...
 
+/* look for sysfs cpu path containing core_id */
+/* such as: sys/bus/cpu/devices/cpu0/topology/core_id */
+func getCoreIdFromCpuBus(cpuBusPath string, threadId int) (int, error) {
+	path := filepath.Join(cpuBusPath, fmt.Sprintf("cpu%d/topology", threadId))
+	file := filepath.Join(path, "core_id")
+
+	num, err := ioutil.ReadFile(file)
+	if err != nil {
+		// report threadId if no NUMA
```
Comment (dashpole) [inline machine/machine.go]: does this failure always indicate there is no NUMA?  Maybe only return threadId if the error is `os.IsNotExist(err)`.

File: machine/machine.go [hunk 2]
```diff
@@ -127,6 +130,53 @@ func parseCapacity(b []byte, r *regexp.Regexp) (uint64, error) {
...
+
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return 0, err
+	}
+
+	nodeId := 0
+	for _, file := range files {
+		filename := file.Name()
+		if !strings.HasPrefix(filename, "node") {
```
[the remark this answers is not in the document]
Comment (dashpole) [reply machine/machine.go]: As in, can we use the regexp to do the matching? I.E. `regexp.MatchString`

File: machine/machine.go (hunk 1 above)
Comment (lubinsz) [reply machine/machine.go]: Done. I have removed it.

File: machine/machine.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (lubinsz) [reply machine/machine.go]: regexp.MatchString was added for it.

File: machine/machine.go (hunk 1 above)
Comment (dashpole) [reply machine/machine.go]: I was suggesting 
```
if os.IsNotExist(err) {
    // report threadId if no NUMA
    return threadId, nil
} else if err != nil {
    return 0, err
}
```
I don't have enough context on the change to know if that is _actually_ the right thing to do here, but whatever you think the correct behavior is works for me.

**Comment by dashpole** [comment]
let me know about the `os.IsNotExist(err)`; otherwise lgtm

**Pull request merged**